sample_gallery/nl2sql/README.MD

```diff
@@ -0,0 +1,110 @@
+
+```bash
+    pip  install  -r  requirements.txt
+```
+
+2. Deploy an Azure SQL Database with sample (AdventureWorksLT) - [Setup Guide)([Create a single database - Azure SQL Database | Microsoft Learn](https://learn.microsoft.com/en-us/azure/azure-sql/database/single-database-create-quickstart?view=azuresql&tabs=azure-portal)
+3. Install and setup the [ODBC Driver for Microsoft SQL Server (17)](https://learn.microsoft.com/en-us/sql/connect/odbc/download-odbc-driver-for-sql-server?view=sql-server-ver16#version-17) to connect to the Azure SQL Database using pyodbc
+	- (Local PromptFlow): Install the driver on the development machine
+	- (Azure AI Studio): The Automatic Runtime doesn't contain the ODBC Driver, so you need to create a custom runtime with promptflow, pyodbc, and the driver installed. [Customize Runtime](https://learn.microsoft.com/en-us/azure/ai-studio/how-to/create-manage-runtime)
+4. (Optional but highly recommended) Install and configure [Prompt flow for VS Code extension](https://marketplace.visualstudio.com/items?itemName=prompt-flow.prompt-flow) follow [Quick Start Guide](https://microsoft.github.io/promptflow/how-to-guides/quick-start.html).
+
+    💡 This extension is optional but highly recommended for flow development and debugging.
+
+### Connections Setup
+
+We need to setup 2 connections to run this sample. [Manage PromptFlow Connections](https://microsoft.github.io/promptflow/how-to-guides/manage-connections.html)
+
+ 1. Azure OpenAI connection: Create a connection to you Azure OpenAI endpoint with a GPT-4 Model deployment. 
+ 2. Custom Connection: Create a custom connection for the Azure SQL Database. Sample YAML
+
+```YML
+$schema: https://azuremlschemas.azureedge.net/promptflow/latest/CustomConnection.schema.json
+name: "to_replace_with_connection_name"
+type: custom
+configs:
+  Server_name: "tcp:<server-name>.database.windows.net,1433"
+  User_name: "<sql-login-name>"
+  Database_name: "AdventureWorksLT"
+secrets:
+# Don't replace the '<user-input>' placeholder. The application will prompt you to enter a value when it runs.
+  Password: "<user-input>"
+```
+
+Once the connections are created, modify the python and LLM nodes in the flow.dag.yaml with the appropriate connections names.
+
+###### get_table_names (python tool)
+
+```YAML
+- name: get_table_names
+    type: python
+    source:
+    type: code
+    path: get_table_names.py
+    inputs:
+    sqlconn: <sql-connection-name>
+    schema_name: ${inputs.schema_name}
+```
+
+###### identify_table_llm (LLM tool)
+
+```YAML
+- name: identify_table_llm
+  type: llm
+  source:
+    type: code
+    path: identify_table_llm.jinja2
+  inputs:
+    deployment_name: gpt-4
+    response_format:
+    type: text
+    user_query: ${inputs.user_query}
+    table_names: ${get_table_names.output}
+  connection: azure_openai_connection
+  api: chat
+```
+
+#### Sample Flow Run (VSCode)
+
```

Comment:
I ran the flow successfully! But during the process, I got 2 errors. Maybe should make some minor refinements or reminders to help users run it smoothly.
(1) I should set the server fjile work rules by adding my IP to enable access
![屏幕截图 2024-04-01 142633](https://github.com/microsoft/promptflow-resource-hub/assets/40321821/f20ade91-9246-4b71-a268-e0fbb937259f)
(2) I should change the deriver name 'ODBC Driver 17' to 'ODBC_Driver 18' uri in 'get_table_schema.py' and 'get_table_names.py' based on the specific version driver I used. Otherwise, I will get this error:
![image](https://github.com/microsoft/promptflow-resource-hub/assets/40321821/cb721df0-b2ae-4bc9-899c-42c420d50c8d)